source/merge.ts

```diff
@@ -27,10 +27,10 @@ export default function merge(target: Options, ...sources: Options[]): Options {
 		}
 	}
 
-	return target;
+	return target as Target & Source;
 }
 
-export function mergeOptions(...sources: Options[]): Options {
+export function mergeOptions(...sources: Partial<Options>[]): Partial<Options> & { hooks: Partial<Hooks> } {
```

Comment:
The `host` property in Options is required but the `mergeOptions` can technically merge parts of the options.